app/src/main/java/com/android/solvit/seeker/ui/provider/SelectProvider.kt

```diff
@@ -916,19 +941,19 @@ fun SelectProviderScreen(
 ) {
   // Lock Orientation to Portrait
   val context = LocalContext.current
+
   DisposableEffect(Unit) {
     val activity = context as? ComponentActivity
     activity?.requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_PORTRAIT
     onDispose {
       locationViewModel.clear()
+
       activity?.requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_UNSPECIFIED
     }
   }
 
   val selectedService by listProviderViewModel.selectedService.collectAsState()
-  if (selectedService != null)
-      listProviderViewModel.filterProviders(
-          filter = { provider -> provider.service == selectedService }, "Service")
+  LaunchedEffect(selectedService) { listProviderViewModel.getProviders() }
```

Comment:
I believe that this modification causes all the providers to appear for a very short amount of time, each time we enter a select provider screen. At least that is what I noticed while testing.

Reply:
Do you an idea on how it can be better implemented ?